DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Utility.cs

```diff
@@ -0,0 +1,172 @@
+                    HasTimeZone = true,
+                };
+            }
+            else
+            {
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, default),
+                    HasTimeZone = false,
+                };
+            }
+        }
+
+        public static string GenerateDicomDateString(DateTimeOffset date)
+        {
+            if (date == null)
+            {
+                return null;
+            }
+
+            return date.ToString("yyyyMMdd", CultureInfo.InvariantCulture);
+        }
+
+        public static string GenerateDicomDateTimeString(DateTimeObject date)
+        {
+            if (date == null)
+            {
+                return null;
+            }
+
+            if (date.HasTimeZone == null || !(bool)date.HasTimeZone)
+            {
+                return date.DateValue.ToString("yyyyMMddhhmmss.ffffff", CultureInfo.InvariantCulture);
+            }
+            else
+            {
+                return date.DateValue.ToString("yyyyMMddHHmmss.ffffffzzz", CultureInfo.InvariantCulture).Replace(":", string.Empty);
+            }
+        }
+
+        public static AgeValue ParseAge(string age)
+        {
+            if (string.IsNullOrEmpty(age))
+            {
+                return null;
+            }
+
+            Dictionary<string, AgeType> ageTypeMapping = new Dictionary<string, AgeType>
+            {
+                { "Y", AgeType.Year },
+                { "M", AgeType.Month },
+                { "W", AgeType.Week },
+                { "D", AgeType.Day },
+            };
+
+            foreach (var item in ageTypeMapping)
+            {
+                if (new Regex(@"\d{3}" + item.Key).IsMatch(age))
+                {
+                    return new AgeValue(uint.Parse(age.Substring(0, 3)), item.Value);
+                }
+            }
+
+            if (uint.TryParse(age, out uint result))
+            {
+                return new AgeValue(result, AgeType.Year);
+            }
+            else
+            {
+                throw new Exception("Invalid age string");
+            }
+        }
+
+        public static string AgeToString(AgeValue age)
+        {
+            Dictionary<string, AgeType> ageTypeMapping = new Dictionary<string, AgeType>
+            {
+                { "Y", AgeType.Year },
+                { "M", AgeType.Month },
+                { "W", AgeType.Week },
+                { "D", AgeType.Day },
+            };
+
+            if (age == null)
+            {
+                return null;
+            }
+
+            foreach (var item in ageTypeMapping)
+            {
+                if (age.AgeType == item.Value)
+                {
+                    if (age.Age.ToString().Length > 3)
+                    {
+                        throw new Exception();
+                    }
+
```

Comment:
why?

Reply:
In DICOM
Date Format : A string of characters of the format YYYYMMDD; where YYYY shall contain year, MM shall contain the month, and DD shall contain the day, interpreted as a date of the Gregorian calendar system.

Reply:
DateTime Format : 

A concatenated date-time character string in the format:

YYYYMMDDHHMMSS.FFFFFF&ZZXX

The components of this string, from left to right, are YYYY = Year, MM = Month, DD = Day, HH = Hour (range "00" - "23"), MM = Minute (range "00" - "59"), SS = Second (range "00" - "60").

FFFFFF = Fractional Second contains a fractional part of a second as small as 1 millionth of a second (range "000000" - "999999").

&ZZXX is an optional suffix for offset from Coordinated Universal Time (UTC), where & = "+" or "-", and ZZ = Hours and XX = Minutes of offset.

The year, month, and day shall be interpreted as a date of the Gregorian calendar system.

A 24-hour clock is used. Midnight shall be represented by only "0000" since "2400" would violate the hour range.

The Fractional Second component, if present, shall contain 1 to 6 digits. If Fractional Second is unspecified the preceding "." shall not be included. The offset suffix, if present, shall contain 4 digits. The string may be padded with trailing SPACE characters. Leading and embedded spaces are not allowed.

A component that is omitted from the string is termed a null component. Trailing null components of Date Time indicate that the value is not precise to the precision of those components. The YYYY component shall not be null. Non-trailing null components are prohibited. The optional suffix is not considered as a component.

A Date Time value without the optional suffix is interpreted to be in the local time zone of the application creating the Data Element, unless explicitly specified by the Timezone Offset From UTC (0008,0201).

UTC offsets are calculated as "local time minus UTC". The offset for a Date Time value in UTC shall be +0000.

Reply:
A string of characters with one of the following formats -- nnnD, nnnW, nnnM, nnnY; where nnn shall contain the number of days for D, weeks for W, months for M, or years for Y.

Example: "018M" would represent an age of 18 months.

Reply:
@qfyin

Reply:
The comments here helped PR reviewer to understand the code.
It would be even better if you put the link to standard in the code comments. That will help future reader/maintainer.

Reply:
Sure, I will add comments in code.